config/routes.rb

```diff
@@ -13,6 +13,9 @@
   # get "/users/:id/posts/:id", to: "posts#post"
 
   resources :users, only: [:show, :index] do
-    resources :posts, only: [:show, :index]
+    resources :posts, only: [:show, :index, :create, :new] do
+      resources :comments, only: [:create, :new]
+      resources :likes, only: [:create]
+      end
   end
 end
```

Comment:
- [ ] [Optional]I do not know why you removed the root route, Kindly make your user index your root route